webui/react/src/pages/ExperimentDetails/ExperimentCheckpoints.tsx

```diff
@@ -0,0 +1,287 @@
+import { FilterDropdownProps, SorterResult } from 'antd/es/table/interface';
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+
+import Badge, { BadgeType } from 'components/Badge';
+import CheckpointModalTrigger from 'components/CheckpointModalTrigger';
+import InteractiveTable, { InteractiveTableSettings } from 'components/InteractiveTable';
+import Section from 'components/Section';
+import { defaultRowClassName, getFullPaginationConfig } from 'components/Table';
+import TableBatch from 'components/TableBatch';
+import TableFilterDropdown from 'components/TableFilterDropdown';
+import { terminalRunStates } from 'constants/states';
+import useModalCheckpointRegister from 'hooks/useModal/Checkpoint/useModalCheckpointRegister';
+import useModalModelCreate from 'hooks/useModal/Model/useModalModelCreate';
+import usePolling from 'hooks/usePolling';
+import useSettings, { UpdateSettings } from 'hooks/useSettings';
+import { getExperimentCheckpoints } from 'services/api';
+import { Determinedcheckpointv1State,
+  V1GetExperimentCheckpointsRequestSortBy } from 'services/api-ts-sdk';
+import { encodeCheckpointState } from 'services/decoder';
+import ActionDropdown from 'shared/components/ActionDropdown/ActionDropdown';
+import { ModalCloseReason } from 'shared/hooks/useModal/useModal';
+import { ErrorLevel, ErrorType } from 'shared/utils/error';
+import { validateDetApiEnum, validateDetApiEnumList } from 'shared/utils/service';
+import {
+  CheckpointState, CoreApiGenericCheckpoint, ExperimentBase,
+} from 'types';
+import handleError from 'utils/error';
+
+import settingsConfig, { Settings } from './ExperimentCheckpoints.settings';
+import { columns as defaultColumns } from './ExperimentCheckpoints.table';
+import css from './ExperimentTrials.module.scss';
+
+interface Props {
+  experiment: ExperimentBase;
+  pageRef: React.RefObject<HTMLElement>;
+}
+
+enum CheckpointAction {
```

Comment:
Just curious about this, aren't typescript enums sort of "soft deprecated" in favor or literal types?

Reply:
This would also let you use string literals like you want to below.

Reply:
I'm open to that. I used an enum here to mimic the pattern in the `ExperimentTrials` page